app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -1581,6 +1580,8 @@ void backToTop() {
   public void onConfigurationChanged(Configuration newConfig) {
     super.onConfigurationChanged(newConfig);
     toggleActionItemsConfig();
+    // Forcing redraw of RecyclerView children so that the tabs are properly oriented on rotation
+    tabRecyclerView.setAdapter(tabsAdapter);
```

Comment:
`tabsAdapter.notifyDataSetChanged` doesn't work?

Reply:
The layouts were not redrawn using this.

Reply:
`tabRecyclerView.invalidate()`?

Reply:
Nope. I tried all of them. :laughing:

Reply:
Then I found this solution on SO.

Reply:
Well it makes me unhappy but that is the day to day work of using android apis.